### PR TITLE
[DO NOT MERGE] Add gender options to State Pension calculator

### PR DESCRIPTION
**Pull request by MuriloDalRi**
Adds non-binary and prefer not to say as gender options to the State
Pension calculator flow, a new outcome for these two options, updates the
calculator to handle these two new options.

[Trello](https://trello.com/c/rSdYn49f/2177-3-state-pension-calculator-update)

:warning: This application is Continuously Deployed: :warning:

- Merged changes are automatically deployed to staging and production.

- Make sure you follow [the guidance for deployments](https://docs.publishing.service.gov.uk/manual/development-pipeline.html#merge-your-own-pull-request) **before** you merge.

- Check your branch is being deployed in the [Release app](https://release.publishing.service.gov.uk/applications/smartanswers), after merging.

- Changes to start pages are **not** continuously deployed, and are [deployed using a different process](https://github.com/alphagov/smart-answers/blob/master/doc/smart-answer-flow-development/publishing.md).

### CODE REVIEW

File: lib/smart_answer/calculators/state_pension_age_calculator.rb
Comment (ChrisBAshton) [inline lib/smart_answer/calculators/state_pension_age_calculator.rb:26]: I see you've added a splat argument here to account for sometimes having no arguments, and sometimes having one (either `:male` or `:female`). I think a default argument would do the job more cleanly:

```suggestion
    def state_pension_age(gender_for_calculation = gender)
```

Then you would replace all the `args.last` calls with `gender_for_calculation`.

**Review by ChrisBAshton — COMMENTED** [review]
Nice one reusing the existing calculation logic 👍  hopefully we can simplify it by removing the splat arguments.

File: lib/smart_answer/calculators/state_pension_age_calculator.rb
Comment (MuriloDalRi) [reply lib/smart_answer/calculators/state_pension_age_calculator.rb]: Thanks, Chris! I tried that and for some reason, it didn't work, probably something else was wrong. I've changed it now!

**Review by ChrisBAshton — APPROVED** [review]
Thanks Murlio! Looks good to me. A couple of small suggestions but non-blocking, so up to you if you want to implement them.

**Comment by theseanything** [comment]
@MuriloDalRi How come we have "Do not merge" in the title? If its waiting for content review/department approval, could you add the relevant label?

**Comment by MuriloDalRi** [comment]
> @MuriloDalRi How come we have "Do not merge" in the title? If its waiting for content review/department approval, could you add the relevant label?

DWP is testing it with users

**Comment by theseanything** [comment]
@MuriloDalRi is there any update on this?

**Comment by MuriloDalRi** [comment]
> @MuriloDalRi is there any update on this?

They've received the results from user research this week so we'll get an update soon.

**Pull request merged**
